File: example_services/scala/finatra/example/example-service/src/main/scala/com/me/example/modules/MessageDaoModule.scala
package com.me.example.modules

import javax.inject.Singleton

import com.me.example.persistence.{MessageDao, SomeRealMessageDao}
import com.google.inject.Provides
import com.twitter.inject.TwitterModule

object MessageDaoModule extends TwitterModule {

  @Singleton
  @Provides
  def providesMessageDao: MessageDao = {
    new SomeRealMessageDao
  }
}

File: example_services/java/finatra/example/example-service/src/main/java/com/me/example/modules/MessageDaoModule.java
package com.me.example.modules;

import com.me.example.persistence.InMemoryMessageDao;
import com.me.example.persistence.MessageDao;
import com.google.inject.Provides;
import com.twitter.inject.TwitterModule;

import javax.inject.Singleton;


public class MessageDaoModule extends TwitterModule {
    @Singleton
    @Provides
    MessageDao providesMessageDao() {
        return new InMemoryMessageDao();
    }
}
